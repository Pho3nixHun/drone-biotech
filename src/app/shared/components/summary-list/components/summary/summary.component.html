<div class="flex flex-row space-x-1">
    <ng-content select="p" />
</div>
