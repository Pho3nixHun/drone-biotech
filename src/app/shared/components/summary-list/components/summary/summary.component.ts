import { Component } from '@angular/core';

@Component({
    selector: 'app-summary',
    imports: [],
    templateUrl: './summary.component.html',
})
export class SummaryComponent {}
