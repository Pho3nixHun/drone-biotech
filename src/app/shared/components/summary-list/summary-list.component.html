<div class="flex flex-col lg:flex-row justify-between">
    <ng-content select="app-summary" />
</div>
