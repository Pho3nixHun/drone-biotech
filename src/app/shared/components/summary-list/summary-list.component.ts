import { Component } from '@angular/core';

@Component({
    selector: 'app-summary-list',
    imports: [],
    templateUrl: './summary-list.component.html',
})
export class SummaryListComponent {}
