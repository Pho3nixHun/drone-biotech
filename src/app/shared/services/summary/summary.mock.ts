import { Summary } from './summary.service.model';

export const summaries: Summary[] = [
    {
        allowedRole: 'customer',
        titleKey: 'DashboardPage.summary.activeOrder.title',
        emoji: 'üìã',
        quantity: 2,
        unitType: null,
        colorType: 'blue',
    },
    {
        allowedRole: 'customer',
        titleKey: 'DashboardPage.summary.totalArea.title',
        emoji: 'üåæ',
        quantity: 145,
        unitType: 'hectare',
        colorType: 'green',
    },
    {
        allowedRole: 'customer',
        titleKey: 'DashboardPage.summary.spending.title',
        emoji: 'üí∞',
        quantity: 126000,
        unitType: 'money',
        colorType: 'purple',
    },
    {
        allowedRole: 'customer',
        titleKey: 'DashboardPage.summary.pending.title',
        emoji: '‚è≥',
        quantity: 3,
        unitType: null,
        colorType: 'yellow',
    },
    {
        allowedRole: 'pilot',
        titleKey: 'DashboardPage.summary.assignedMissions',
        emoji: 'üéØ',
        quantity: 2,
        unitType: null,
        colorType: 'blue',
    },
    {
        allowedRole: 'pilot',
        titleKey: 'DashboardPage.summary.completed',
        emoji: '‚úÖ',
        quantity: 8,
        unitType: null,
        colorType: 'green',
    },
    {
        allowedRole: 'pilot',
        titleKey: 'DashboardPage.summary.flight.title',
        emoji: 'üöÅ',
        quantity: 47.5,
        unitType: 'hour',
        colorType: 'purple',
    },
    {
        allowedRole: 'pilot',
        titleKey: 'DashboardPage.summary.performance.title',
        emoji: '‚≠ê',
        quantity: 4.8,
        unitType: 'rating',
        colorType: 'yellow',
    },
    {
        allowedRole: 'office',
        titleKey: 'DashboardPage.summary.activeMission.title',
        emoji: 'üéÆ',
        quantity: 12,
        unitType: null,
        colorType: 'blue',
    },
    {
        allowedRole: 'office',
        titleKey: 'DashboardPage.summary.availablePilots.title',
        emoji: 'üë®‚Äç‚úàÔ∏è',
        quantity: 5,
        unitType: null,
        colorType: 'green',
    },
    {
        allowedRole: 'office',
        titleKey: 'DashboardPage.summary.unassigned.title',
        emoji: '‚ö†Ô∏è',
        quantity: 3,
        unitType: null,
        colorType: 'red',
    },
    {
        allowedRole: 'office',
        titleKey: 'DashboardPage.summary.revenue.title',
        emoji: 'üíº',
        quantity: 12450,
        unitType: 'money',
        colorType: 'purple',
    },
];
