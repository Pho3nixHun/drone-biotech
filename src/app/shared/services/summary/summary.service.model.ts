import { UserRole } from '@stores/auth/auth.model';

export type SummaryColorType = 'blue' | 'green' | 'purple' | 'yellow' | 'red';
export type SummaryUnitType = 'hectare' | 'money' | 'rating' | 'hour';

export interface Summary {
    allowedRole: UserRole;
    emoji: string;
    titleKey: string;
    quantity: number;
    colorType: SummaryColorType;
    unitType: SummaryUnitType | null;
}
