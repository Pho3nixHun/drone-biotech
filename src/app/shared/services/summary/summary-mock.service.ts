import { Injectable, signal } from '@angular/core';
import { UserRole } from '@stores/auth/auth.model';
import { Observable, of, map } from 'rxjs';
import { Summary } from './summary.service.model';
import { SummaryService } from './summary.service';

@Injectable({
    providedIn: 'root',
})
export class SummaryMockService {
    public getSummaries = (role: UserRole): Observable<Summary[]> =>
        of(summaries()).pipe(
            map((summaries) =>
                summaries.filter((summary) => summary.allowedRole === role)
            )
        );
}

const summaries = signal<Summary[]>([]);

export const updateSummaries = (obj: Summary[]) => summaries.set(obj);

export const provideSummaryMockService = () => ({
    provide: SummaryService,
    useClass: SummaryMockService,
});
