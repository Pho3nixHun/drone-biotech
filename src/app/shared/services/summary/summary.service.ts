import { Injectable } from '@angular/core';
import { map, Observable, of } from 'rxjs';
import { summaries } from './summary.mock';
import { UserRole } from '@stores/auth/auth.model';
import { Summary } from './summary.service.model';

@Injectable({
    providedIn: 'root',
})
export class SummaryService {
    public getSummaries = (role: UserRole): Observable<Summary[]> =>
        of(summaries).pipe(
            map((summaries) =>
                summaries.filter((summary) => summary.allowedRole === role)
            )
        );
}
