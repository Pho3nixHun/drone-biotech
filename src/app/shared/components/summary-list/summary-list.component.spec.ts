import { ComponentFixture, TestBed } from '@angular/core/testing';

import { SummaryListComponent } from './summary-list.component';
import { SummaryComponent } from './components/summary/summary.component';
import { Component } from '@angular/core';

@Component({
    imports: [SummaryListComponent, SummaryComponent],
    template: `
        <app-summary-list>
            <app-summary />
        </app-summary-list>
    `,
})
class TestHostComponent {}

describe('SummaryListComponent', () => {
    let fixture: ComponentFixture<TestHostComponent>;
    let compiled: HTMLElement;

    beforeEach(async () => {
        await TestBed.configureTestingModule({
            imports: [TestHostComponent],
        }).compileComponents();

        fixture = TestBed.createComponent(TestHostComponent);
        compiled = fixture.debugElement.nativeElement;
    });

    //Snapshot test
    it('should render the template', () => {
        //Arrange
        /*No need for arrange*/

        //Act
        fixture.detectChanges();
        //Assert
        expect(compiled).toMatchSnapshot();
    });
});
