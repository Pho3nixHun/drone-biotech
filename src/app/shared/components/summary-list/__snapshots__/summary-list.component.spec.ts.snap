// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`SummaryListComponent should render the template 1`] = `
<div
  id="root0"
  ng-version="20.1.7"
>
  <app-summary-list>
    <div
      class="flex flex-col lg:flex-row justify-between"
    >
      <app-summary>
        <div
          class="flex flex-row space-x-1"
        />
      </app-summary>
    </div>
  </app-summary-list>
</div>
`;
